packages/hash/api/src/graphql/typeDefs/knowledge/user.typedef.ts

```diff
@@ -0,0 +1,47 @@
+import { gql } from "apollo-server-express";
+
+export const userTypedef = gql`
+  extend type Query {
+    me(linkResolveDepth: Int!, linkTargetEntityResolveDepth: Int!): Subgraph!
+    """
+    Determines whether a provided shortname is already taken
+    """
+    isShortnameTaken(shortname: String!): Boolean!
```

Comment:
Any reason why this is a resolver rather than a subgraph filtering on the clientside?

Reply:
What do you mean? Have the FE fetch all orgs and all users, and then check whether the shortname exists on any of them?

I don't think this can be a longterm approach because:
1. the payload size of the sub-graph would be large in production instances
2. a non-signup user probably shouldn't be able to simply fetch all the users in a HASH instance
3. there is additional validation going on under the hood to determine whether a shortname is "taken", because we have a list of restricted shortnames we have on the BE that we may want instance admins to be able to add to down the line.

Reply:
It is true that we'd have to consider non-authorized users in the case of client-side filtering, yeah. I think the size argument is quite relevant as well. I suppose this endpoint is also helpful for signing up and testing whether or not a shortname is taken

Reply:
> I suppose this endpoint is also helpful for signing up and testing whether or not a shortname is taken

To my knowledge that's the only place it's used right now